lib/ui/common/ActionSheet.dart

```diff
@@ -0,0 +1,193 @@
+                  topRight: actionStyle.borderRadius
+              )
+          ),
+          child: HeaderAndFooterListView.builder(
+              physics: NeverScrollableScrollPhysics(),
+              shrinkWrap: true,
+              itemCount: viewModel.listOfActions.length,
+              itemBuilder: (BuildContext context, int index) =>
+            buildActionCell(actionStyle, viewModel.listOfActions[index], viewModel.listOfActions.length, index),
+            header: buildDropLine(actionStyle),
+            footer: RoundedButton.build(style: ActionSheetLargeRoundedButtonStyle(), context: context, title: viewModel.buttonTitle)
+          )
+    ));
+  }
+
+  static Widget buildDropLine(ActionSheetStyle actionStyle) {
+    return Container(
+      height: actionStyle.dropLineHeight,
+      //FIXME: when margin property is setted as actionStyle.dropLineEdge is crashing
+      margin: EdgeInsets.only(left: 160, right: 160, top: 8, bottom: 50),
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.all(actionStyle.dropLineRadius),
+        color: actionStyle.dropLineColor,
+      ),
+    );
+  }
+
+  static Widget buildActionCell(ActionSheetStyle actionStyle, CustomAction action, int numberOfActions, int currentAction) {
+    return Column(
+      children: <Widget>[
+        Card(
+          elevation: actionStyle.cardElevation,
+          color: actionStyle.transparentColor,
+          child: InkWell(
+            onTap: () => print(action.text),
+            child: Container(
+              padding: actionStyle.cardEdge,
+              alignment: Alignment.center,
+              height: actionStyle.cardHeight,
+              child: Wrap(
+                direction: Axis.horizontal,
+                crossAxisAlignment: WrapCrossAlignment.start,
+                children: _buildActionContent(actionStyle, action),
+              ),
+            ),
+          ),
+        ),
+        currentAction == numberOfActions-1 ?  Wrap() : ListDivider.build(),
+      ],
+    );
+  }
+
+  static List<Widget> _buildActionContent(ActionSheetStyle actionStyle, CustomAction action) {
+    List<Widget> listBuilder = [];
+    if (action.primaryIcon != null && action.selectionIcon == null) {
+      listBuilder.add(
+      Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        children: <Widget>[
+          Image.asset(action.primaryIcon, height: actionStyle.smallIconHeight),
+          SizedBox(width: actionStyle.iconLineSpace),
+          Text(action.text, style: action.isHighlighted ? actionStyle.highlightTextStyle : actionStyle.mainTextStyle)
+        ],
+      ));
+
+    } else if (action.selectionIcon != null) {
+      listBuilder.add(
+      Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: <Widget>[
+          Row(children: <Widget>[
+            Image.asset(action.primaryIcon, height: actionStyle.bigIconHeight),
+            SizedBox(width: actionStyle.iconLineSpace),
```

Comment:
looks like repeated code here? can we use variables to remove the repeated code?

what's small icon height and big icon height?

Reply:
Now it's rebuilded without repeating code, also it's only a iconHeight which size change with the style.